data-table/table.js

```diff
@@ -687,33 +686,100 @@ function createRowDeleteBtn(config) {
+    wrapperDiv.appendChild(JSONBtn);
 }
 
-    /**
-// eslint-disable-next-line no-unused-vars
-/** function toJSON() {
-     * Parses data held in HTML to JSON and sends it to client
-     * TODO: Fill this out
-     * TODO: Implement Serialization
-} */
+/**
+ * Returns the HTML element corresponding to a cell at a specific index
+ * @param {object} config           - Table configuration object
+ * @param {Number} row              - Row on which the cell is located
+ * @param {Number} column           - Column in which the cell is located
+ * @returns {HTMLTableCellElement}  - HTML element of a specific cell
+ */
+function getCellElement(config, row, column) {
+    return document.getElementById(cellIndexToElementId(config.wrapperDivId, row, column))
+}
+
+/**
+ * Gets a 2D array containing all data from all fields of all cells in the table
+ * @param {object} config   - Table configuration object
+ * @returns {*[]}           - 2D array of cell objects
+ */
+function getTableData(config) {
+    let tableData = [];
+    for (let row = 0; row < config.numRows; row++) {
+        tableData.push(getRowData(config, row));
+    }
+    return tableData;
+}
+
+/**
+ * Gets a 1D array containing all data from all fields of all cells in a specific row of the table
+ * @param {object} config   - Table configuration object
+ * @param {Number} row      - Specific row to pull data from
+ * @returns {*[]}           - 1D array of cell objects
+ */
+function getRowData(config, row) {
+    let rowData = [];
+    for (let col = 1; col < config.numColumns; col++) {
+        rowData.push(getCellData(config, row, col));
+    }
+    return rowData
+}
+
+/**
+ * Gets an object containing all data from all fields of a specific cell
+ * @param {object} config   - Table configuration object
+ * @param {Number} row      - Row index for the cell
+ * @param {Number} col      - Column index for the cell
+ * @returns {object}        - Object containing cell data in the following format: {[fieldName]:[fieldValue],...}
+ */
+function getCellData(config, row, col) {
+    let cellData = {};
+    let cell = getCellElement(config, row, col);
+    let index = 0;
+
+    /*
+    Assumes that each cell has only field labels as children
+     */
+    cell.childNodes.forEach(function (label) {
+        /*
+        Assumes that each label has two children. The first is
+        text containing the label's name and the second is the
+        space the user can interact with.
+         */
+        let value = "";
+        switch (config.datumConfig.types[index]) {
+            case Number:
+                value = parseInt(label.childNodes[1].value, 10) || config.datumConfig.values[index];
+                break;
+            case Boolean:
+                value = label.childNodes[1].checked;
+                break;
+            default:
+                value = label.childNodes[1].value;
+        }
+        cellData[config.datumConfig.names[index].toLowerCase()] = value;
```

Comment:
why toLowerCase?

Reply:
toLowerCase means that we can have a field with a label `Value` and that same field will be accessed in the JSON object by `JSONobject.data.value` instead of `JSONobject.data.Value`. The latter looked weird to me so I went for all lowercase.

Marking this as resolved for now and adding a FIXME so we remember to check in about this later.